### ISSUE TITLE
Debian package

### ISSUE THREAD

**Issue by sharkwouter** [lutris/lutris#2553]
I was wondering if it would be possible to include Lutris into Debian, but I found a ton of work on this has already been done. @stephanlachnit updated the Debian package big time in #2476 and the ITP bug in Debian [was recently updated](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=754129#44) as well. There is still a [TODO file](https://github.com/lutris/lutris/blob/master/debian/TODO), but I couldn't find a bug report.

The TODO file currently contains the following items:
- review copyright file and remove the copyright.old for the svgs
- create man-page for lutris
- create lintian-override or man-page for lutris-wrapper

I would like to help out on this if possible, is any help needed? I think the first item has probably been resolved with #2542. Is there any documentation on what lutris wrapper does?

**Comment by sharkwouter** [comment]
Just to add some information, I am not a Debian contributor. I just have some experience making deb packages and I've read the new maintainers guide.

**Comment by stephanlachnit** [comment]
As you said I already tried to get the package into Debian. I filled an [RFS](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=945588), but it looks like no DD or DM is currently interested in sponsoring (or even reviewing) the package. I'll still "maintain" it though, with the source on [salsa](https://salsa.debian.org/stephanlachnit-guest/lutris) and the dsc on [mentors](https://mentors.debian.net/package/lutris).

You could look into the man pages, that would be the the easiest thing to do here in upstream, but actually it's not that important for an official upload.

The major issue is still the copyright, that's why I leaft it in the TODO file. I checked everything as good as I could but without a review from the [debian-legal](https://lists.debian.org/debian-legal/) mailing list I don't see high chances for an official upload (copyright file could be fine but lutris still not DFSG compliant). Tbh I haven't asked on the mailing for a review though, you could do that if you want.

**Comment by stephanlachnit** [comment]
So I have some good and some bad news considering the an official Debian package. @JohnScott623 did a [review](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=945588#14), and the svgs are not DFSG-compliant. However, here's the good news: that doesn't block an upload, Debian could just ship the package without the svgs ([explanation](https://wiki.debian.org/DebianMentorsFaq#What_does_.2BIBw-dfsg.2BIB0_or_.2BIBw-ds.2BIB0_in_the_version_string_mean.3F)). So overall, here's a little to-do list:

More important stuff:
- [ ] Lutris has to handle missing svgs without showing an error
- [ ] Remove `copyright.old` and entry to review it in the `debian/TODO` file
- [ ] Don't ship `python-distro` (see review)

Less important stuff:
- [ ] Add man page for `lutris`
- [ ] Add man page for `lutris-wrapper`, create a reasonable lintian override, or move it to a location for helper-executables (maybe `/usr/lib/lutris` or `/usr/share/lutris/bin/`)
- [ ] Remove the shipped winetricks file and rely on system-installation as fallback instead (add it to recommends), or provide a full winetricks including a symlink to `usr/bin` (add it to provides&breaks)

@strycore what are your thoughts about removing the shipped winetricks? If I remember it right, Lutris is using it only as a fallback anyway, so it wouldn't make a real difference for anyone.
And do you have some documentation for the `lutris-wrapper` executable?

**Comment by amurzeau** [comment]
Hi,

It seems you try to make the Debian packaging inside the lutis git repository.
When I packaged streamlink for Debian, it had already contained a debian directory with outdated files.

I found easier to just have a separate git repository managed by myself to do the packaging stuff as packaging has sometimes different requirements than upstream development (especially when stable release is approaching). This avoided round trips as streamlink developpers are more centered to streamlink itself than linux packaging.

So I suggest you keep Debian packaging in a separate git directory so you can remove stuff you don't need (like winetricks, python-dist, "GNU/Linux") and adjust stuff for Debian without involving upstream lutris.
Sometimes, changes made to the packaged application are better to be made upstream, sometimes not.
For example, one reason to keep bundled stuff in lutris current distribution is that lutris is only tested with one version of these (and upgrading winetricks or python might cause bugs) and its easier to manage to avoid too many issues for users.

**Comment by stephanlachnit** [comment]
> It seems you try to make the Debian packaging inside the lutis git repository.
> When I packaged streamlink for Debian, it had already contained a debian directory with outdated files.

Thanks for you concern. The Debian package requires an own git repository anyway ([here it is](https://salsa.debian.org/stephanlachnit-guest/lutris)), for example it needs an own changelog, a changed control file and so on.
It's not about punching all changes necessary for Debian into upstream, rather it's about improving the upstream Ubuntu package as well because the Debian package requires the improvements anyway. Don't forget, Ubuntu imports packages from unstable/testing _directly_ into their devel branch, overriding changes made specific to the Ubuntu package or an Ubuntu-only upload, simply because they should be same (with very few exceptions).
An examples for these improvements would be the machine-readable copyright or the clean file I created anyway. A man-page isn't just useful for Debian users.

> For example, one reason to keep bundled stuff in lutris current distribution is that lutris is only tested with one version of these (and upgrading winetricks or python might cause bugs) and its easier to manage to avoid too many issues for users.

I highly doubt that. The bundled winetricks is a fallback, as Lutris bundles winetricks afaik with the custom wine version anyway ([relevant code](https://github.com/lutris/lutris/blob/master/lutris/runners/commands/wine.py#L342)).
I already added an option to use your system winetricks in #2494, now it's just about replacing the default fallback option to the system installation, which is a way more reasonable fallback option.
Before I updated winetricks in #2544, it was quite old (autumn 2018). As a fallback, that's okay, but if it isn't updated regularly (= maintenance work), you better stick with the version bundled from your distro. If you don't use a Ubuntu version older than 19.04 (which is weird for a gaming setup in my opinion anyway), the version in Debian and Ubuntu was newer, and they probably do a better job to check for compatibility. Maybe that was different when winetricks was first bundled within Lutris, I don't know, but I think nowadays it doesn't make that much sense.

I haven't look into why python-distro is bundled and where it's actually installed during the installation, but I don't think it's necessary. It is a package which literally manages distro specific stuff for you, so it would be kinda weird if you have to ship it with your program.

One can argue about the missing svgs, but since Linux is not just about the kernel but also about open source and freedom, I think it's fair to let users (or in this case an entire distribution) get rid of _svgs_ with a non-free copyright (its not something essential anyway). I don't except every program to be completely free, but an open source program could help out where it doesn't hurt. The changes necessary to not show an error or at least not crash are probably very small, haven't looked into it.
Same goes for the "for GNU/Linux" line. I don't want to force anyone to officially support Hurd or BSDs (I literally couldn't care less about these kernels), but _if_ there is a working version which doesn't require anything more than pressing the compile button, you can change stuff like "for GNU/Linux" to "developed for GNU/Linux" or "for Unix-like systems", or just remove it completely. Doesn't hurt anyone, and if other people are willing to do the effort for you there is no reason to block it.

Long story short: all the changes directly benefit upstream or other upstream adopters without adding any regressions or tons of code. I don't try to turn upstream into the Debian package (that would be kinda disrespectful), but a more diverse package is at least not bad for anyone, if not even better for everyone.

**Comment by amurzeau** [comment]
Indeed, I pretty much agree with everything you said given Lutris is targeted to linux (and maybe Hurd/BSD latter), and all distros should have python and winetricks available.

**Comment by strycore** [comment]
First and most importantly: can we stop making a big deal about SVG files? If there are problematic files with unclear copyright then I'd rather open up Inkscape and spend 15 minutes drawing an alternative than going through painful legal BS. The complaint I received about the use of Flaticons seems invalid to me. The license forbids from distributing their whole icon set, which is totally understandable. We don't do that and only distribute a few selected icons, with their attribution. There is a deep misunderstanding in what the license says but I also understand that the Flaticon license is not an open source license and thus might cause issues for inclusion in Debian. Again, we can provide other icons, let's not make this a problem.

lutris-wrapper should not have a manpage. It is exclusively used by lutris and never meant to be run by a user. We can put this file in another location, if we write the code to support it.

I don't know what the status of python-distro is in every distro we support. This is used as a replacement for the now deprecated `platform.linux_distribution` (see https://docs.python.org/2/library/platform.html#platform.linux_distribution )

I'm all in favor of removing mentions of "GNU/Linux" but really good luck running Lutris on anything else, just see the file `lutris/utils/linux.py` to see how much Linux specific stuff there is.

We can remove winetricks, there is no reason for us to be shipping it in the client, it's a runtime component.

Can we please make the changes needed for Debian packaging as small as possible compared to upstream? I have been packaging Lutris for Ubuntu and Debian since the beginning of the project, this is why there is a `debian` folder in our source repository. I have to continue maintaining those packages until Lutris is distributed widely, which will take a while. I'm willing to make the changes needed to make Lutris DFSG compliant.

**Comment by stephanlachnit** [comment]
> First and most importantly: can we stop making a big deal about SVG files? If there are problematic files with unclear copyright then I'd rather open up Inkscape and spend 15 minutes drawing an alternative than going through painful legal BS. The complaint I received about the use of Flaticons seems invalid to me. The license forbids from distributing their whole icon set, which is totally understandable. We don't do that and only distribute a few selected icons, with their attribution. There is a deep misunderstanding in what the license says but I also understand that the Flaticon license is not an open source license and thus might cause issues for inclusion in Debian. Again, we can provide other icons, let's not make this a problem.

I personally don't care about weather those icons are DFSG-compliant or not, I haven't even read the DFSG, and if I probably don't have enough legal knowledge to understand it anyway. My only concern is to get this package DFSG-compliant.
I think we should just ask over at the [debian-legal](https://lists.debian.org/debian-legal/) mailing list, after all they know the best if the svgs can live in the archive. If they decide that there are problems with some svgs, and you're willing to redraw them, that would be great. I don't know if it's just about the license, or also about copyrighted logos.

> lutris-wrapper should not have a manpage. It is exclusively used by lutris and never meant to be run by a user. We can put this file in another location, if we write the code to support it.

Perfect, will look into it.

> I don't know what the status of python-distro is in every distro we support. This is used as a replacement for the now deprecated `platform.linux_distribution` (see https://docs.python.org/2/library/platform.html#platform.linux_distribution )

I took a look now and Lutris uses [lutris/vendor/distro.py](https://github.com/lutris/lutris/blob/master/lutris/vendor/distro.py) version v1.3.0, but there are not many changes since v1.2.0 except more distros.
I compared it to several distros via [pkgs.org](https://pkgs.org/download/python-distro) or their package sites:
 * Ubuntu: v1.3.0 since 18.10
 * Debian: v1.3.0 since 10
 * Solus: couldn't find an online search site
 * Fedora: v1.3.0 since 29
 * openSUSE: v1.2.0 since 15.0
 * Arch/Manjaro: v1.4.0 obviously
 * Clear Linux: inside python-extras, but I couldn't find the bundled version
 * Mageia: v1.4.0 since 7
 * Gentoo: installer is for v1.0.4, but Gentoo always compiles upstream afaik
 * Slackware: v1.4.0 since 14.2
 * CentOS: v1.2.0 since 7

python-distro can also be installed via pip (package name is `distro`), so I think it's pretty same to assume that we can use the system version. The only change necessary is to use `from distro import xy` instead of `from lutris.vendor.distro import xy`.

> I'm all in favor of removing mentions of "GNU/Linux" but really good luck running Lutris on anything else, just see the file `lutris/utils/linux.py` to see how much Linux specific stuff there is.

Oh I didn't see that. I'll guess the "just press the compile button" claim was invalid then.

> We can remove winetricks, there is no reason for us to be shipping it in the client, it's a runtime component.

Cool, and you even already done it. Should we add winetricks to recommends (or at least suggests), as it is used as fallback? It would also take into account the winetricks dependencies.

> Can we please make the changes needed for Debian packaging as small as possible compared to upstream? I have been packaging Lutris for Ubuntu and Debian since the beginning of the project, this is why there is a `debian` folder in our source repository. I have to continue maintaining those packages until Lutris is distributed widely, which will take a while.

Yes I think it's the best to keep differences as small as possible. I haven't packed Lutris after 0.5.4 so I can't tell exactly what needs to be changed but except the changelog it's probably just a couple of lines to make clear that it's uploaded for Debian and not Ubuntu.

> I'm willing to make the changes needed to make Lutris DFSG compliant.

Thanks! As I said, I can't say what needs to be changed, but I think we'll get there eventually. Removing the svgs in the Debian package was just a "quick and dirty" idea to get the package at least uploaded, one can still argue about it later.

**Comment by stephanlachnit** [comment]
I asked for a review on debian-legal, here is the thread: https://lists.debian.org/debian-legal/2020/01/msg00020.html

**Comment by ulm** [comment]
> I asked for a review on debian-legal, here is the thread: https://lists.debian.org/debian-legal/2020/01/msg00020.html

Just as a cross-reference, there is now also a Gentoo bug for this issue: https://bugs.gentoo.org/706432

**Comment by stephanlachnit** [comment]
Here's what's necessary to build it on Debian: https://salsa.debian.org/stephanlachnit-guest/lutris/-/commit/4c050548311c679b0d30ab739f26f7f599cce725
The diff is basically zero :) (except the changelog ofc)
Let's see if someone will sponsor it, I'll ask around.

**Comment by stephanlachnit** [comment]
@strycore coming back to the Flaticon license: I can only guess, but I think Lutris is still in NEW because no-one wants to check whether the Flaticon license is DFSG compliant. Since you wrote this back then:

> First and most importantly: can we stop making a big deal about SVG files? If there are problematic files with unclear copyright then I'd rather open up Inkscape and spend 15 minutes drawing an alternative than going through painful legal BS. The complaint I received about the use of Flaticons seems invalid to me. The license forbids from distributing their whole icon set, which is totally understandable. We don't do that and only distribute a few selected icons, with their attribution. There is a deep misunderstanding in what the license says but I also understand that the Flaticon license is not an open source license and thus might cause issues for inclusion in Debian. Again, we can provide other icons, let's not make this a problem.
[...]
Can we please make the changes needed for Debian packaging as small as possible compared to upstream? I have been packaging Lutris for Ubuntu and Debian since the beginning of the project, this is why there is a debian folder in our source repository. I have to continue maintaining those packages until Lutris is distributed widely, which will take a while. I'm willing to make the changes needed to make Lutris DFSG compliant.

If you could really re-create the svgs under a free license (CC0-1.0 would be the best for this I guess), that would be very much appreciated. I understand if you don't want to invest the time though.

**Comment by djibux** [comment]
Hello.

Are those the files in question?
```
Files: share/lutris/icons/hicolor/symbolic/apps/nintendogamecube-symbolic.svg
 share/lutris/icons/hicolor/symbolic/apps/nintendones-symbolic.svg
 share/lutris/icons/hicolor/symbolic/apps/nintendosnes-symbolic.svg
 share/lutris/icons/hicolor/symbolic/apps/segadreamcast-symbolic.svg
Copyright: freepik <https://www.flaticon.com/authors/freepik>
License: Flaticon
```

If so, the following icons are CC0: https://www.svgrepo.com/vectors/nintendo/
I can submit a PR.

**Comment by stephanlachnit** [comment]
I think there are some more but rhese are definitely some of them, thanks!
A PR would be nice, you can cc me.

**Comment by djibux** [comment]
I submitted a PR replacing all Flaticon icons with CC0 equivalent.  You are right, there were a few other.

**Comment by stephanlachnit** [comment]
Nice, no more Flaticon license problems. Once #2808 is merged Lutris should be lintian clean.

**Comment by strycore** [comment]
Nice! Should this stay open until the package is uploaded somewhere?

**Comment by stephanlachnit** [comment]
If you wouldn't mind, I would leave it open until it's in Debian, but the original issue of this is solved.

**Comment by strycore** [comment]
I'm closing issues. This one goes. I'm sure we won't forget about Debian packaging when the time comes.

**Comment by stephanlachnit** [comment]
Okay, finally heard back from an FTP masters team (for version 0.5.5 xd). The package was rejected - surprisingly due to the `CC-BY-NC-SA-2.0` license of RetroPie logo. NC (i. e. non-commercial) is violation (at least in version 2.0) to the DFSG since it restricts the usage of the software.
Now honestly I couldn't even find where that logo is from. I found an svg here (https://github.com/RetroPie/RetroPie-Docs/blob/master/docs/images/logo.svg), and the repo is under GPLv3, but I'm not sure if they actually intended the icon to be licensed under GPLv3 as well. I'll ask them how it is licensed and will report back.

**Comment by stephanlachnit** [comment]
@strycore @muzena what exactly happend in 0.5.7.1?
It's `arch: any` again and it uses an outdated way of specifying the debhelper version. I don't see the point here?
Alos, the debian folder got moved out. This is a bit annoying for me now, I'm sure there would have been a less "dramatic" way.

**Comment by muzena** [comment]
@stephanlachnit 
#3024 #3026

**Comment by strycore** [comment]
I made the job easier for Debian maintainers to review the package.

**Comment by soredake** [comment]
Any progress on this?

**Comment by stephanlachnit** [comment]
> Any progress on this?

Yes, looking forward for the upcoming 0.5.8, right now it should land in Debian. I will talk a bit about the state in Debian on the upcoming MiniDebConfOnline: https://mdco2.mini.debconf.org/talks/1-overview-of-gaming-utilities-in-debian/

**Comment by stephanlachnit** [comment]
Lutris 0.5.8 is in NEW: https://ftp-master.debian.org/new/lutris_0.5.8-1.html

Luckily NEW is quite empty rn :)

**Comment by strycore** [comment]
That is awesome!

**Comment by stephanlachnit** [comment]
Update: Lutris needs to go in contrib, but I don't that's much of a problem considering that most ppl have Contrib / Multiverse for Steam anyway. Will reupload asap.

**Comment by soredake** [comment]
https://packages.ubuntu.com/hirsute/multiverse/lutris
https://packages.debian.org/sid/lutris

**Comment by sharkwouter** [comment]
Awesome, congratulations on getting this done! I'll close this issue then.
